libzfs.pyx

```diff
@@ -626,29 +626,22 @@ cdef class ZFS(object):
                 ZFS.__datasets_snapshots(handle, <void*>snap_list)
                 data[name]['snapshots'] = snap_list[1:]
 
-            for top_level_prop in configuration_data['top_level_props']:
-                data[name][top_level_prop] = properties.get(top_level_prop, {}).get('value')
-
-                if top_level_prop == 'mountpoint' and data[name][top_level_prop] == 'none':
-                    data[name]['mountpoint'] = None
+            data[name]['mountpoint'] = None
+            if properties.get('mounted', {}).get('value') == 'yes':
```

Comment:
If we are retrieving properties selectively, and consumer says `mountpoint`, he/she would only get `None` even if there's a valid mountpoint being retrieved because we are relying on another property `mounted` which is implicit.

Reply:
He should not use top-level `mountpoint` property in that case

Reply:
We have usages relying on that behaviour, should we update them or should we keep the same behaviour as we had before ?

Reply:
Which usages? I've only found one, it's in jails_freebsd.py and it does not read mountpoint at all

Reply:
We have one in `zfs.dataset.locked_datasets` as well

Reply:
@sonicaj I am rewriting this method completely